test/functional/unified-spec-runner/operations.ts

```diff
@@ -234,17 +234,20 @@ operations.set('find', async ({ entities, operation }) => {
 
 operations.set('findOneAndReplace', async ({ entities, operation }) => {
   const collection = entities.getEntity('collection', operation.object);
-  return collection.findOneAndReplace(operation.arguments.filter, operation.arguments.replacement);
+  const { filter, replacement, ...opts } = operation.arguments;
+  if (opts.returnDocument) {
+    opts.returnDocument = opts.returnDocument.toLowerCase();
+  }
```

Comment:
This suggestion could be overly defensive here but can we make a `processOptions` or `translateOptions` function in `unified-utils.ts` and use it in both places here? I can foresee us needing more of these spec-to-driver translations that might be good to box up the functionality in one place. 

You'll find `patchDbOptions` and `patchCollectionOptions` in that utils file that currently do nothing since we're lucky to share camel-casing and correct value types with the spec but I can see when those hooks already exist would come in handy.

Reply:
let me know if the added code is what you had in mind

Reply:
Looks good!